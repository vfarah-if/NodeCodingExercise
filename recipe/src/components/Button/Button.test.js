import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import {
	PrimaryButtonStory,
	DefaultButtonStory,
	SmallButtonStory,
	LargeButtonStory,
} from './Button.stories';

describe('button', () => {
	test('should create a button with a primary Button display name', () => {
		const { container } = render(
			<PrimaryButtonStory {...PrimaryButtonStory.args} />
		);

		expect(screen.getByRole('button')).toHaveTextContent('Button');
		expect(container.querySelector('button')).toMatchSnapshot();
	});

	test('should create a button simple Button display name', () => {
		const renderResponse = render(
			<DefaultButtonStory {...DefaultButtonStory.args} />
		);

		expect(
			renderResponse.container.querySelector('button')
		).toMatchSnapshot();
	});

	test('should create a button large Button display name', () => {
		const renderResponse = render(
			<LargeButtonStory {...LargeButtonStory.args} />
		);

		expect(
			renderResponse.container.querySelector('button')
		).toMatchSnapshot();
	});

	test('should create a button small Button display name', () => {
		const renderResponse = render(
			<SmallButtonStory {...SmallButtonStory.args} />
		);

		expect(
			renderResponse.container.querySelector('button')
		).toMatchSnapshot();
	});
});
