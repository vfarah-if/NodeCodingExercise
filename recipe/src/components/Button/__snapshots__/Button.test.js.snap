// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should create a button large Button display name 1`] = `
<button
  class="storybook-button storybook-button--large storybook-button--secondary"
  type="button"
>
  Button
</button>
`;

exports[`button should create a button simple Button display name 1`] = `
<button
  class="storybook-button storybook-button--medium storybook-button--secondary"
  type="button"
>
  Button
</button>
`;

exports[`button should create a button small Button display name 1`] = `
<button
  class="storybook-button storybook-button--small storybook-button--secondary"
  type="button"
>
  Button
</button>
`;

exports[`button should create a button with a primary Button display name 1`] = `
<button
  class="storybook-button storybook-button--medium storybook-button--primary"
  type="button"
>
  Button
</button>
`;
